controllers/openstackdataplanerole_controller.go

```diff
@@ -264,6 +264,11 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
 		if err != nil || !isReady {
 			return ctrl.Result{}, err
 		}
+	} else {
+		// We don't require any baremetal nodes, so we will remove these conditions from the status
+		instance.Status.Conditions.Remove(dataplanev1.RoleBareMetalProvisionReadyCondition)
+		instance.Status.Conditions.Remove(dataplanev1.RoleDNSDataReadyCondition)
+		instance.Status.Conditions.Remove(dataplanev1.RoleIPReservationReadyCondition)
```

Comment:
Removing these is cleaner, and matches how we handle similar situations in the openstack-operator:
https://github.com/openstack-k8s-operators/openstack-operator/blob/main/pkg/openstack/heat.go#L31

Reply:
I personally prefer keeping these to make it more explicit that these steps are skipped but this is jut my own preference.

Reply:
I originally did, but it seemed a little less clean, possibly confusing for users to have things there that they didn't ask for?
https://github.com/openstack-k8s-operators/dataplane-operator/compare/d648e9123b760cc7604f163d2182be8cfa4e31de..d8a00a56bfd98b0610f21310665193bfbae8cf6b